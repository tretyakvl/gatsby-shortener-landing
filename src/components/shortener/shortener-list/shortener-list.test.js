import React from 'react'
import renderer from 'react-test-renderer'
import ShortenerList from './shortener-list'

describe('ShortenerList', () => {
  it('renders correctly with zero items', () => {
    const tree = renderer
      .create(<ShortenerList />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with three items', () => {
    const props = {
      shortenedLinks: [
        {
          original: 'https://www.google.com',
          shortened: 'https://www.google.com'
        },
        {
          original: 'https://www.amazon.com',
          shortened: 'https://www.amazon.com'
        },
        {
          original: 'https://www.google.ru',
          shortened: 'https://www.google.ru'
        }
      ]
    }
    const tree = renderer
      .create(<ShortenerList {...props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
