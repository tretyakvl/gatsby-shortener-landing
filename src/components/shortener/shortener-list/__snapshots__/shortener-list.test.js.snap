// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortenerList renders correctly with three items 1`] = `
<ul
  className="shortener__list"
>
  <li
    className="shortener__item"
    style={Object {}}
  >
    <span>
      https://www.google.com
    </span>
    <a
      href="https://www.google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://www.google.com
    </a>
    <button
      className="button-module button square thin400"
      onClick={[Function]}
    >
      copy
    </button>
  </li>
  <li
    className="shortener__item"
    style={Object {}}
  >
    <span>
      https://www.amazon.com
    </span>
    <a
      href="https://www.amazon.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://www.amazon.com
    </a>
    <button
      className="button-module button square thin400"
      onClick={[Function]}
    >
      copy
    </button>
  </li>
  <li
    className="shortener__item"
    style={Object {}}
  >
    <span>
      https://www.google.ru
    </span>
    <a
      href="https://www.google.ru"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://www.google.ru
    </a>
    <button
      className="button-module button square thin400"
      onClick={[Function]}
    >
      copy
    </button>
  </li>
</ul>
`;

exports[`ShortenerList renders correctly with zero items 1`] = `
<ul
  className="shortener__list"
/>
`;
